enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,329 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Create a new e2e testing framework for the new agent based installer project. 
+Currently, for this project there would be potentially two existing e2e frameworks 
+that could be used, dev-scripts and assisted-test-infra. However, both of these have 
+significant challenges in re-purposing them for this new project. 
+
+The main challenge with either of these existing frameworks is that the environment
+setup, installation and subsequent tests they run are all highly coupled together.
+With the ephemeral agent installer image it is looking like parts of the potential
+e2e flow is in both of these frameworks. However, there is no way to mix and match the
+things we need to simply use them without significant re-factoring effort of both
+of these frameworks. 
+
+A more expected challenge is that some of the routines it is looking like we will 
+need are completely new and thus I am taking the time to argue that it will be less
+risky to create a new framework for the ephemeral agent installer image.
+
+These challenges will be explained in more detail in the User Stories section below.
+
+## Motivation
+
+The new agent based installer project will need an e2e testing framework for validation.
+We should look at the existing frameworks and understand what re-purposing them will 
+entail for this new project.
+
+This enhancement looks at the existing e2e frameworks dev-scripts and assisted-test-infra 
+closely since this is the previous work in this space. The motivation is to expand on
+what we have learned from those frameworks try to come up with a solution that we can 
+reuse for new installers without these challenges in the future.
+
+### Goals
+
+* This proposal will focus using this new e2e framework in the new ephemeral agent based 
+installer image project.
+* After discussing with the Metal Platform team we will be improving parts of dev-scripts
+that will be used for the ephemeral agent installer. The scope of these improvements will
+be limited.
+
+### Non-Goals
+
+* Re-factoring dev-scripts and assisted-test-infra for reasons not related to the new 
+ephemeral agent based installer image project.
+* There should be some overlap in the pieces that are refactored for use in the new project.
+However it is still a non-goal for outside installer to use this new framework in their e2e 
+tests for the upcoming release. We will plan to create a point of convergence after the
+ephemeral agent based installer project is out of prototyping and MVP phase.
+
+## Proposal
+
+Create an e2e testing framework for the ephemeral agent based installer image.
+
+### User Stories
+
+#### Run Assisted Installer Outside k8s
+
+As a developer, I need to run Assisted Installer without installing a k8s platform first 
+so that I can test the installation of a users first cluster, cluster 0.
```

Comment:
Is this user story still relevant? It feels very focused on an implementation detail, unlike the other stories.

Reply:
I feel like this implementation detail is important to point out. It cuts to the root of the chicken/egg problem that we have with the existing Assisted Installer.